### PR TITLE
Add recipe for SLEEF

### PR DESCRIPTION
**Pull request by bluescarni**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by bluescarni** [comment]
@isuruf I'd need some advice on how to proceed with the openmp dependency. There are 2 issues:

- in order to avoid warning/errors by conda, I had to manually add ``_openmp_mutex`` as a dependency (otherwise conda would complain that the package is using openmp features but does not declare a dependency on the openmp runtime). Is this the correct way of proceding?
- the build system of the package fails to detect openmp on osx. Do you have any advice/experience about enabling openmp in osx?

**Comment by bluescarni** [comment]
@isuruf I think that now the openmp setup in the recipe is correct. Would you mind giving a review?

File: recipes/sleef/bld.bat [hunk 1]
```diff
@@ -0,0 +1,11 @@
+mkdir build
+cd build
+
+cmake ^
+    -G"Visual Studio 15 2017" ^
```
Comment (isuruf) [inline recipes/sleef/bld.bat:5]:
```suggestion
```

File: recipes/sleef/bld.bat [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    -DBUILD_TESTS=no ^
```
Comment (isuruf) [inline recipes/sleef/bld.bat:6]:
```suggestion
    -DBUILD_TESTS=no ^
    -DCMAKE_BUILD_TYPE=Release ^
```

File: recipes/sleef/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  url: https://github.com/shibatch/sleef/archive/{{ version }}.tar.gz
+  sha256: 415ee9b1bcc5816989d3d4d92afd0cd3f9ee89cbd5a33eb008e69751e40438ab
+  patches:
+    - install_dll.diff  # [win]
+
+build:
+  number: 0
+  run_exports:
+    # SO name changes at minor rev bumps.
+    - {{ pin_subpackage('sleef', max_pin='x.x') }}
```
Comment (isuruf) [inline recipes/sleef/meta.yaml:17]: Isn't the SONAME `libsleef.so.3`?
```suggestion
    - {{ pin_subpackage('sleef', max_pin='x') }}
```

Comment (bluescarni) [reply recipes/sleef/meta.yaml:17]: It is, but I am not 100% sure what their ABI stability policy is. I thought I was just being overly conservative via an ``x.x`` pin. Does the ``x`` pin mean that version 3.n is considered compatible with 3.m?

**Comment by bluescarni** [comment]
@isuruf do you think this is good to go or does this need more changes?

**Pull request merged**